pages/page.py

```diff
@@ -2,126 +2,68 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import time
-
 from selenium.common.exceptions import NoSuchElementException
-from selenium.common.exceptions import ElementNotVisibleException
-from selenium.common.exceptions import StaleElementReferenceException
-from selenium.common.exceptions import TimeoutException
-from selenium.common.exceptions import WebDriverException
-from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support.ui import WebDriverWait as Wait
+
+
+TIMEOUT = 10
 
 
 class Page(object):
-    """
-    Base class for all Pages
-    """
+
+    _page_title = None
+    _url = None
 
     def __init__(self, base_url, selenium):
-        """
-        Constructor
-        """
         self.base_url = base_url
         self.selenium = selenium
-        self.timeout = 10
-        self._selenium_root = hasattr(self, '_root_element') and self._root_element or self.selenium
+        self.timeout = TIMEOUT
 
     @property
     def is_the_current_page(self):
-        WebDriverWait(self.selenium, 10).until(lambda s: self.selenium.title)
-        assert self._page_title == self.selenium.title
+        if self._page_title is not None:
+            Wait(self.selenium, self.timeout).until(lambda s: s.title)
+            assert self._page_title == self.selenium.title
         return True
 
     @property
-    def current_page_url(self):
-        return self.selenium.current_url
+    def url(self):
+        if self._url is not None:
+            return self._url.format(base_url=self.base_url)
+        return self.base_url
 
-    def is_element_present(self, *locator):
-        self.selenium.implicitly_wait(0)
+    def is_element_visible(self, locator):
         try:
-            self._selenium_root.find_element(*locator)
-            return True
-        except NoSuchElementException:
+            return self.selenium.find_element(*locator).is_displayed()
+        except (NoSuchElementException,):
```

Comment:
Well, I believe the parentheses are there because this is a tuple, and I also believe that a tuple needs the comma, so I think this is actually more correct than just `(NoSuchElementException)` which Python actually interprets as just a variable inside parens. It would probably be better to do away with the parens entirely as they are what is unnecessary, but I also know that without them there, if the time comes when we need to add another exception to the list, we would likely add it without the parens and that would be wrong.

So, to sum up, we don't need the parens and they should be removed, *but* removing them increases the chance of a bug being introduced later if we want to add another exception, *therefore* if we are going to keep them we should code it correctly as a tuple which means adding the comma.

What do you think?